src/Symfony/Component/Messenger/Exception/HandlerFailedException.php

```diff
@@ -49,4 +49,19 @@ public function getNestedExceptions(): array
     {
         return $this->exceptions;
     }
+
+    public function hasNestedExceptionOfClass(string $exceptionClassName): bool
+    {
+        return 0 < \count($this->getNestedExceptionOfClass($exceptionClassName));
+    }
+
+    public function getNestedExceptionOfClass(string $exceptionClassName): array
+    {
+        return array_filter(
+            $this->exceptions,
+            function ($exception) use ($exceptionClassName) {
+                return \get_class($exception) === $exceptionClassName;
```

Comment:
should this be an exact match, or an `is_a()` check?
you feature request looks like an `instanceof` check. This would mean `is_a()` here.

Reply:
Ok good catch. In my mind it was a `is_a` check but I should admin an `instanceof` looks better indeed.

Reply:
Lets use `is_a()` instead.

Reply:
please add some related test cases also.